assets/src/dashboard/app/views/myStories/index.js

```diff
@@ -57,6 +63,38 @@ function MyStories() {
     fetchStories({ status });
   }, [fetchStories, status]);
 
+  const setPageSize = useCallback(() => {
+    const { innerWidth } = window;
+
+    if (innerWidth <= theme.breakpoint.raw.min) {
+      _setPageSize({
+        width: theme.grid.min.itemWidth,
+        height: theme.grid.min.imageHeight,
+      });
+    } else if (innerWidth <= theme.breakpoint.raw.mobile) {
+      _setPageSize({
+        width: theme.grid.mobile.itemWidth,
+        height: theme.grid.mobile.imageHeight,
+      });
+    } else if (innerWidth <= theme.breakpoint.raw.tablet) {
+      _setPageSize({
+        width: theme.grid.tablet.itemWidth,
+        height: theme.grid.tablet.imageHeight,
+      });
+    } else {
+      _setPageSize({
+        width: theme.grid.desktop.itemWidth,
+        height: theme.grid.desktop.imageHeight,
+      });
+    }
+  }, []);
+
+  useEffect(() => {
+    const unsubscribe = window.addEventListener('resize', setPageSize);
+    setPageSize();
+    return unsubscribe;
+  }, [setPageSize]);
```

Comment:
This is to ensure the grid sizing is responsive. UnitsProvider requires the correct page size to correctly size the elements.

Reply:
Correct. This is something we've wrestled with, but the `UnitsProvider` is a much easier approach then relative units which require a much more nuanced understanding how to correctly construct paddings, font sizes, etc.

Reply:
Yeah this works fine, but I might extract this out to a hook as to not clutter up `myStories` view

Reply:
IMHO this useEffect and the const setPageSize = useCallback above can be combined neater into useResizeEffect for the actual container. You will automatically get the dimensions of the container that you can use for calculations. You just need to make sure that the container itself is not scrollable - i.e. it'd be a wrapper over your scroller. You can use thus obtained container with to do the math and set the page size.